serial/rfc2217.py

```diff
@@ -621,7 +621,7 @@ def read(self, size=1):
                     raise SerialException('connection failed (reader thread died)')
                 buf = self._read_buffer.get(True, timeout.time_left())
                 if buf is None:
-                    return bytes(data)
+                    raise SerialException('connection failed (connection failed or lost)')
```

Comment:
Are you sure that buf is None only when an error occurred? How is a normal TCP close from the remote end handled?

Reply:
Looking at https://github.com/labgrid-project/pyserial/blob/master/serial/rfc2217.py the `put` operations on the Queue object are pretty straightforward:
`self._read_buffer.put(None)` occurs on
* [connection failure](https://github.com/labgrid-project/pyserial/blob/master/serial/rfc2217.py#L767)
* [lost connection](https://github.com/labgrid-project/pyserial/blob/master/serial/rfc2217.py#L770)